insights/client/connection.py

```diff
@@ -1086,19 +1119,16 @@ def checkin(self):
         url = self.inventory_url + "/hosts/checkin"
         logger.debug("Sending check-in request to %s with %s" % (url, canonical_facts))
         try:
-            response = self.session.post(url, headers={"Content-Type": "application/json"}, data=json.dumps(canonical_facts))
-            # Change to POST when the API is fixed.
-        except REQUEST_FAILED_EXCEPTIONS as exception:
-            _api_request_failed(exception)
+            response = self.post(url, headers={"Content-Type": "application/json"}, data=json.dumps(canonical_facts))
+        except RuntimeError as e:
+            logger.error(e)
             return None
```

Comment:
We may consider raising an exception in case of failure instead of returning success. That would save another conditional and still correctly result in setting process status code to _constants.sig_kill_bad_.